src/app/campaign/store/effects/campaign.effects.ts

```diff
@@ -25,34 +25,124 @@ export class CampaignEffects {
+      } else {
+        campaignSaveResult = this.campaignService.createCampaign(payload.campaign);
+      }
+      return campaignSaveResult.pipe(
+        mergeMap((campaignDoc: HalDoc) => {
+          const deletedFlights: { [id: number]: Observable<HalDoc> } = payload.deletedFlights.reduce(
+            (acc, flight) => ({ ...acc, [flight.id]: this.campaignService.deleteFlight(flight.id) }),
+            {}
+          );
+          return payload.deletedFlights.length
+            ? forkJoin(deletedFlights).pipe(map(deletedFlightDocs => ({ campaignDoc, deletedFlightDocs })))
+            : of({ campaignDoc });
+        }),
+        mergeMap(({ campaignDoc, deletedFlightDocs }: { campaignDoc: HalDoc; deletedFlightDocs: { [id: number]: HalDoc } }) => {
+          const updatedFlights: { [id: number]: Observable<HalDoc> } = payload.updatedFlights.reduce(
+            (acc, flight) => ({ ...acc, [flight.id]: this.campaignService.updateFlight(flight) }),
+            {}
+          );
+          return payload.updatedFlights.length
+            ? forkJoin(updatedFlights).pipe(map(updatedFlightDocs => ({ campaignDoc, deletedFlightDocs, updatedFlightDocs })))
+            : of({ campaignDoc, deletedFlightDocs });
         }),
-        map(({ campaignDoc }: { campaignDoc: HalDoc }) => new actions.CampaignFormSaveSuccess({ campaignDoc })),
-        catchError(error => of(new actions.CampaignFormSaveFailure({ error })))
+        mergeMap(
+          ({
+            campaignDoc,
+            updatedFlightDocs,
+            deletedFlightDocs
+          }: {
+            campaignDoc: HalDoc;
+            updatedFlightDocs: { [id: number]: HalDoc };
+            deletedFlightDocs: { [id: number]: HalDoc };
+          }) => {
+            const createdFlights: { [id: number]: Observable<HalDoc> } = payload.createdFlights.reduce(
+              (acc, flight) => ({ ...acc, [flight.id]: this.campaignService.createFlight(flight) }),
+              {}
+            );
+            return payload.createdFlights.length
+              ? forkJoin(createdFlights).pipe(
+                  map(createdFlightDocs => ({ campaignDoc, updatedFlightDocs, createdFlightDocs, deletedFlightDocs }))
+                )
+              : of({ campaignDoc, updatedFlightDocs, deletedFlightDocs });
+          }
+        ),
+        tap(
+          ({
+            campaignDoc,
+            deletedFlightDocs,
+            updatedFlightDocs,
+            createdFlightDocs
+          }: {
+            campaignDoc: HalDoc;
+            deletedFlightDocs: { [id: number]: HalDoc };
+            updatedFlightDocs: { [id: number]: HalDoc };
+            createdFlightDocs: { [id: number]: HalDoc };
+          }) => {
+            this.toastr.success('Campaign saved');
+            // navigate to newly created campaign/flight or away from deleted flight
+            const campaignId = campaignDoc.id;
+            const flightId = +this.router.url.split('/flight/').pop();
```

Comment:
Could also have used the router store state to get the routed flight id or included it in the action, but you know what, this is fine (url.split.pop() taken from the component that was routing on save previously.)